### ISSUE TITLE
"value-no-vendor-prefix" doesn't catch {white-space: -webkit-pre-wrap;} etc

### ISSUE THREAD

**Issue by gaz7** [stylelint/stylelint#2203]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I'm running stylelint over legacy code, and it failed to catch vendor prefixed pre-wrap values.
It catches other vendor prefixed values such as `a { display: -webkit-flex; }`, so it seems to be specific to pre-wrap.

I'm guessing that pre-wrap may never have required vendor prefixes, which is why the test doesn't work. However, it would be really useful if stylelint could still catch issues like this.

> Which rule, if any, is this issue related to?

"value-no-vendor-prefix"

> What CSS is needed to reproduce this issue?

```css
.wrapword {
    white-space: -moz-pre-wrap !important;  /* Mozilla, since 1999 */
    white-space: -pre-wrap;      /* Opera 4-6 */
    white-space: -o-pre-wrap;    /* Opera 7 */
    white-space: pre-wrap;       /* css-3 */
    white-space: -webkit-pre-wrap; /* Newer versions of Chrome/Safari */
    white-space: normal;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
    "extends": "stylelint-config-standard",
    "rules": {
        "indentation": 4,
        "block-no-empty": null,
        "at-rule-no-vendor-prefix": true,
        "media-feature-name-no-vendor-prefix": true,
        "property-no-vendor-prefix": true,
        "selector-no-vendor-prefix": true,
        "value-no-vendor-prefix": true
    }
}
```

> Which version of stylelint are you using?

7.6.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

gulp-stylelint

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

not that I'm aware. The file is SCSS, but the code that it's failing to catch is standard CSS

> What did you expect to happen?

Expected a warning

> What actually happened (e.g. what warnings or errors you are getting)?

No warnings

**Comment by evilebottnawi** [comment]
@gaz7 check you `browserlist` (http://caniuse.com/#search=pre-wrap), `value-no-vendor-prefix` rule throw error only for values, which meets the `browserlist` condition.

**Comment by jeddy3** [comment]
@gaz7 Thank you for the report and for using the issue template.

> I'm guessing that pre-wrap may never have required vendor prefixes, which is why the test doesn't work. However, it would be really useful if stylelint could still catch issues like this.

I'm not sure this is viable as we [rely on](https://github.com/stylelint/stylelint/blob/master/lib/utils/isAutoprefixable.js) `autoprefixer` to determine whether a vendor prefix is needed or not i.e. we need a way to determine if a value is a standard one unnecessarily prefixed or a proprietary one that is correctly prefixed.

Take this example (albeit for _properties_):

```css
a {
  -webkit-overflow-scrolling: smooth;
  -webkit-border-radius: 5px;
}
```

```
3:3 error Unexpected vendor-prefix "-webkit-border-radius" (property-no-vendor-prefix)
```

Here only `-webkit-border-radius` is correctly flagged as `-webkit-overflow-scrolling` is a proprietary property that should be prefixed.

I'm not sure it's feasible to try to distinguish between known proprietary values and unknown ones. 

I'm going to label this as a documentation issue. I think adding the following to the rule description will suffice:

"This rule will only flag prefixed standard values, not prefixed proprietary or unknown ones."

**Comment by gaz7** [comment]
Thanks for getting back to me.

Certainly adding the documentation will help, as I didn't realise the rule only applied to values that autoprefixer can handle.

Would you consider a second mode that the vendor prefix rules could be configured to run under? 
In a strict mode, ALL vendor prefixes would be flagged, and stylelint-disable comments could be used for proprietary properties/values when needed.

**Comment by evilebottnawi** [comment]
I think we can add option `all` (maybe best name), where ignore `all` prefixes.

**Comment by jeddy3** [comment]
Perhaps this specific use case should be a plugin. It'd be easy to build (using `postcss-value-parser` and `postcss.vendor.unprefixed()`).

**Comment by evilebottnawi** [comment]
@jeddy3 maybe should be put to a plugin, if we get another request for such behavior, it can be considered as part of the core, in theory rule have name `value-no-vendor-prefix` and actually a bit misleading, as it is expected that it will report about all prefixes (and has optional support `autoprefixer`), now right name should be `value-no-vendor-prefix-autoprefixer` (or something like). I think it would be right initially to build a rule that tells about all the prefixes and has optional support `autoprefixer` (we can do this in `8.0.0`).

**Comment by jeddy3** [comment]
> maybe should be put to a plugin

Agreed as it's in-line with [our criteria](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#criteria-for-inclusion). The current behaviour of `value-no-vendor-prefix` is _useful to the majority of users_, and I think should remain that way.

@gaz7 Feel free to [create a plugin](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/plugins.md) for your particular use-case. You should only need a few lines of code.
